code/game/machinery/doors/firedoor.dm

```diff
@@ -442,6 +442,19 @@
 	..()
 	icon_state = "frame[constructionStep]"
 
+/obj/structure/firelock_frame/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	if(the_rcd.mode == RCD_DECONSTRUCT)
+		return list("mode" = RCD_DECONSTRUCT, "delay" = 50, "cost" = 16)
+	return FALSE
+
+/obj/structure/firelock_frame/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	switch(passed_mode)
```

Comment:
```suggestion
	if(passed_mode == RCD_DECONSTRUCT)
```

No need for a 1 item switch

Reply:
Also make sure to change the line below it

Reply:
It appears that making the suggested change to firedoor.dm launches me down a rabbit hole of endless errors that don't seem to go away no matter what I do.

Reply:
What exactly are you changing it to?
Line 452 should be struck, line 451 changed to the above change, and then line 453-455 should be de-indented by one indent

Reply:
So lines 451 through 456 would become

```
	If(passed_mode == RCD_DECONSTRUCT)
		to_chat(user, "<span class='notice'>You deconstruct [src].</span>")
		qdel(src)
		return TRUE
	return FALSE

Reply:
... well, this is _absolutely_ spooky. I made those exact changes before and they didn't work, but now they do. Weird.